apps/desktop-wallet/src/components/TokenBadge.tsx

```diff
@@ -16,46 +16,61 @@ You should have received a copy of the GNU Lesser General Public License
 along with the library. If not, see <http://www.gnu.org/licenses/>.
 */
 
-import { Asset } from '@alephium/shared'
 import styled, { css } from 'styled-components'
 
-import useToken, { isFT, isNFT } from '@/api/apiDataHooks/useToken'
+import useFetchToken, { isFT, isNFT } from '@/api/apiDataHooks/useFetchToken'
 import Amount from '@/components/Amount'
 import AssetLogo from '@/components/AssetLogo'
+import SkeletonLoader from '@/components/SkeletonLoader'
+import { TokenId } from '@/types/tokens'
 
-export interface AssetBadgeStyleProps {
-  simple?: boolean
+export interface TokenBadgeStyleProps {
   withBorder?: boolean
   withBackground?: boolean
-  hideNftName?: boolean
   className?: string
+  showSymbol?: boolean
+  showNftName?: boolean
+  showAmount?: boolean
 }
 
-interface AssetBadgeProps extends AssetBadgeStyleProps {
-  assetId: Asset['id']
+interface TokenBadgeProps extends TokenBadgeStyleProps {
+  tokenId: TokenId
   amount?: bigint
+  isLoading?: boolean
 }
 
-const AssetBadge = ({ assetId, amount, simple, hideNftName, className }: AssetBadgeProps) => {
-  const { data: token } = useToken(assetId)
+const TokenBadge = ({ tokenId, className, ...props }: TokenBadgeProps) => {
+  const { data: token } = useFetchToken(tokenId)
 
-  const tooltipContent = isFT(token) || isNFT(token) ? token.name : assetId
+  const tooltipContent = isFT(token) || isNFT(token) ? token.name : tokenId
 
   return (
-    <div className={className} data-tooltip-id="default" data-tooltip-content={tooltipContent}>
-      <AssetLogo tokenId={assetId} size={20} />
-      {isNFT(token) && !hideNftName ? (
-        <AssetSymbol>{token.name}</AssetSymbol>
-      ) : isFT(token) && amount !== undefined ? (
-        <Amount tokenId={assetId} value={amount} />
-      ) : (
-        !simple && isFT(token) && <AssetSymbol>{token.symbol}</AssetSymbol>
-      )}
-    </div>
+    <TokenBadgeStyled className={className} data-tooltip-id="default" data-tooltip-content={tooltipContent}>
+      <AssetLogo tokenId={tokenId} size={20} />
+
+      <TokenBadgeText tokenId={tokenId} {...props} />
+    </TokenBadgeStyled>
   )
 }
 
-export default styled(AssetBadge)`
+const TokenBadgeText = ({ tokenId, amount, isLoading, showNftName, showAmount, showSymbol }: TokenBadgeProps) => {
+  const { data: token } = useFetchToken(tokenId)
+
+  if (isNFT(token) && showNftName) return <TokenSymbol>{token.name}</TokenSymbol>
+
+  if (isFT(token) && isLoading) return <SkeletonLoader height="20px" />
```

Comment:
So we don't show the loader for NFTs?

Reply:
No because we don't need to load balances for NFTs.

Reply:
Of course but is the name readily available through useFetchToken?

Reply:
Good call! I implemented a fix in 267e849a5

Reply:
Nice, thanks! This shows why the "Fetch" prefix can be useful!